src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1210,6 +1212,89 @@ void PhaseIterGVN::optimize() {
   NOT_PRODUCT(verify_PhaseIterGVN();)
 }
 
+#ifdef ASSERT
+void PhaseIterGVN::verify_optimize() {
+  if (is_verify_Value()) {
+    ResourceMark rm;
+    Unique_Node_List worklist;
+    bool failure = false;
+    // BFS all nodes, starting at root
+    worklist.push(C->root());
+    for (uint j = 0; j < worklist.size(); ++j) {
+      Node* n = worklist.at(j);
+      failure |= verify_node_value(n);
+      // traverse all inputs and outputs
+      for (uint i = 0; i < n->req(); i++) {
+        if (n->in(i) != nullptr) {
+          worklist.push(n->in(i));
+        }
+      }
+      for (DUIterator_Fast imax, i = n->fast_outs(imax); i < imax; i++) {
+        worklist.push(n->fast_out(i));
+      }
+    }
+    // If we get this assert, check why the reported nodes were not processed again in IGVN.
+    // We should either make sure that these nodes are properly added back to the IGVN worklist
+    // in PhaseIterGVN::add_users_to_worklist to update them again or add an exception
+    // in the verification code above if that is not possible for some reason (like Load nodes).
+    assert(!failure, "Missed optimization opportunity in PhaseIterGVN");
+  }
+}
+
+// Check that if type(n) == n->Value(), return true if we have a failure
+// We have a list of exceptions, see comments in code.
```

Comment:
May be enumerate list of exceptions and short description.

Reply:
Ok, I can do that. I will just make a short list, and a quick sentence per item. The detailed description I will still leave further down, at the block that actually implements the exception, ok?

Reply:
did that.